### DISCUSSION TITLE
Request for help for setting up alarms using govmomi

### DISCUSSION THREAD

**Discussion by sasikeda** [vmware/govmomi#2782] [Community Help] [answered]
Hi Team,

I am trying to use govmomi for my day-to-day operations and looking for an example that can help set an alarm on datastore. Can someone help share any example related it.

**Comment by embano1** [comment, marked as answer]
Hi, 

do you also want to create the alarm or just `enable` it?

**Reply by sasikeda** [reply]
Thank you so much for your prompt response.
I wanted to perform `GetAlarm`, check its state, if it is not enabled then enable.

Using your PR, I am able to get the status by `info.Info.AlarmSpec.Enabled`. I guess I will have to use `types.EnableAlarm` to set it. But a bit confused with `This` and `Entity`. Can you help share any snippet for its usage?

**Reply by embano1** [reply]
Sure, and I understand the API is not super easy to use.

The respective API was added in vSphere versions after, but **not** including, 6.7.x.

```go
_, err = methods.EnableAlarm(ctx, client, &types.EnableAlarm{
		This:  *am, // AlarmManager instance
		Alarm: a, // the alarm managed object reference you want to enable
        // not sure if you need this too, since the alarm already has it - if needed, 
        // you can reference this field from the alarm information
		// Entity: types.ManagedObjectReference{}, 
})
```

**Reply by embano1** [reply]
API source: https://vdc-download.vmware.com/vmwb-repository/dcr-public/8946c1b6-2861-4c12-a45f-f14ae0d3b1b9/a5b8094c-c222-4307-9399-3b606a04af55/vim.alarm.AlarmManager.html#enableAlarm

**Reply by embano1** [reply]
Let me know if this resolves your question so we can mark this discussion as resolved.

**Reply by sasikeda** [reply]
Thank you so much for the prompt response. It's solved my need.

**Comment by embano1** [comment]
I created an example of how to use the alarm manager to retrieve alarm information: https://github.com/vmware/govmomi/pull/2784

**Reply by sasikeda** [reply]
Thank you so much for the example. It is helpful.
